### PR TITLE
Set empty string as value when setting invalid value multiple times

### LINKED ISSUE #756: Value is wrong after setting an invalid value twice
### Description
The first time the value of a combo box is set to an invalid value, it is reset to `""`. The second time, the invalid value remains

### Steps to reproduce
```
var cb = document.createElement("vaadin-combo-box");
document.body.appendChild(cb);
cb.items = [];
cb.value=5;
console.log(cb.value); // Prints ""
cb.value=5;
console.log(cb.value); // Prints "5"
```

### Browsers Affected

- [X] Chrome
- [X] Firefox
- [X] Safari
- [X] Edge
- [X] IE 11
- [X] iOS Safari
- [X] Android Chrome

### PR DESCRIPTION
**Pull request by samiheikki**
Fixes #756

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-combo-box/757)

### CODE REVIEW

File: src/vaadin-combo-box-mixin.html [hunk 1]
```diff
@@ -789,11 +789,17 @@
...
       const valueIndex = this._indexOfValue(value, this.filteredItems);
+      const previouslySelectedItem = this.selectedItem;
+
       this.selectedItem = valueIndex >= 0
         ? this.filteredItems[valueIndex]
         : (this.dataProvider && this.selectedItem === undefined)
           ? undefined
           : null;
+
+      if (this.selectedItem === null && previouslySelectedItem === this.selectedItem) {
```
Comment (tomivirkki) [inline src/vaadin-combo-box-mixin.html]: Can you cover these conditions with tests also? Or remove any conditions that isn't justified. The test in this pr currently passes fine without them.

Comment (samiheikki) [reply src/vaadin-combo-box-mixin.html]: Done.

**Comment by samiheikki** [comment]
> The Polymer 3 build is now fixed on master, please rebase to get build green.

Rebased and tests are now passing

File: src/vaadin-combo-box-mixin.html (hunk 1 above)
Comment (tomivirkki) [reply src/vaadin-combo-box-mixin.html]: The tests still pass without the first condition `this.selectedItem === null`!

Comment (samiheikki) [reply src/vaadin-combo-box-mixin.html]: Fixed now @tomivirkki

**Pull request merged**